src/custom/hooks/web3.ts

```diff
@@ -30,11 +30,12 @@ export function useEagerConnect() {
 
       // if there is no last saved provider set tried state to true
       if (!latestProvider) {
-        setTried(true)
-      }
-
-      // check if the last saved provider is Metamask
-      if (latestProvider === WalletProvider.INJECTED) {
+        // Try to auto-connect to the injected wallet
+        activate(injected, undefined, true).catch(() => {
```

Comment:
Just out of curiosity, is there a reason why we would try to connect to injected if there is no latestProvider?

Reply:
i answer u in our call, but now u see this fix https://github.com/gnosis/cowswap/issues/1830 then u get a stronger answer :)